SU2_CFD/src/solvers/CTurbSSTSolver.cpp

```diff
@@ -1024,3 +1032,47 @@ void CTurbSSTSolver::SetUniformInlet(const CConfig* config, unsigned short iMark
   }
 
 }
+
+/*--- We determine the interior node that is closest to the wall node. If there is no
+      interior node, we have to take the closest wall node. ---*/
+su2double  CTurbSSTSolver::GetNearest_Neighbor(CGeometry *geometry, unsigned long iPoint, unsigned short iMarker, unsigned long iVertex) {
+  su2double dist_min;
+  su2double distance = 0.0;
+  unsigned long Point_Normal, jPoint;
+
+        /*--- Compute closest normal neighbor, note that the normal are oriented inwards ---*/
+        Point_Normal = 0;
+        // we use distance
+        dist_min = 0.0;
+        for (size_t iNeigh = 0; iNeigh < geometry->nodes->GetnPoint(iPoint); iNeigh++) {
+          jPoint = geometry->nodes->GetPoint(iPoint, iNeigh);
+
+          su2double distance2 = GeometryToolbox::SquaredDistance(nDim,
+                                                             geometry->nodes->GetCoord(iPoint),
+                                                             geometry->nodes->GetCoord(jPoint));
+          // Take the interior node that is closest to the wall node.
+          if ((geometry->nodes->GetViscousBoundary(jPoint) == false) && (distance < dist_min)) {
+            Point_Normal = jPoint;
+            dist_min = sqrt(distance2);
+          }
+        }
+
+
+        if (Point_Normal==0) {
+          cout << "no point found at i = " << iPoint << endl;
+          su2double Area = 0.0;
+          su2double TwoVol = 2.0* (geometry->nodes->GetVolume(iPoint) + geometry->nodes->GetPeriodicVolume(iPoint));
+          for (size_t iNeigh = 0; iNeigh < geometry->nodes->GetnPoint(iPoint); ++iNeigh) {
+            size_t iEdge = geometry->nodes->GetEdge(iPoint, iNeigh);
+            size_t jPoint = geometry->nodes->GetPoint(iPoint, iNeigh);
+            const su2double* Normal = geometry->edges->GetNormal(iEdge);
+            Area = GeometryToolbox::Norm(nDim, Normal);
+
+            dist_min += TwoVol / Area;
```

Comment:
@pcarruscag I now did this. If I compare this length scale in the corner node to the ones next to it, it is much larger.

Reply:
Sure but have you tried what I suggested? Namely, wall distance (not Euclidean), and volume divided by the sum of vertex areas (vertices of viscous markers).
Vertices are on boundaries, edges are internal.

Reply:
Sorry, I'm getting old and am a bit hard of hearing :-) 
OK, so in the code above we only take the vertex areas, we check if jPoint is actually on the wall. using GetViscousBoundary
For the 'regular' routine we can do x.n/||n||
I have modified it, length scales seem to be fine now.

Reply:
Cool thanks for testing. I think it's simpler than that, I'll push something in the weekend, don't start updating tests yet.